### PR TITLE
Bump scalatest, make scala 3 compatible (Ready for review)

### PR DESCRIPTION
**Pull request by hughsimpson**
~The main change here is bumping scalatest, however there are a few other small changes. The common theme is that, once https://github.com/whisklabs/docker-it-scala/pull/146 (or similar version bump pr) is published, the project will build with scala 3~

Cross-compile with scala 3 should now work with artifacts available in maven central. The  docker-testkit-impl-docker-java  dependency is a beta, but since it's just a test dependency I don't think that should be a blocker

### CODE REVIEW

File: src/test/scala/com/redis/SerializationSpec.scala [hunk 1]
```diff
@@ -25,16 +26,18 @@ class SerializationSpec extends FunSpec
 
     r.hmget("hash", "field1", "field2") should be(Some(Map("field1" -> "1", "field2" -> "2")))
 
-    implicit val parseInt = Parse[Int](new String(_).toInt)
+    {
```
Comment (hughsimpson) [inline src/test/scala/com/redis/SerializationSpec.scala:29]: without these brackets, the earlier call to r.hmset will error in scala 3.0.2 with a message about an incorrect forwards-reference to parseInt

File: src/test/scala/com/redis/SerializationSpec.scala [hunk 2]
```diff
@@ -51,10 +54,13 @@ class SerializationSpec extends FunSpec
   it("should use a provided implicit formatter") {
     case class Upper(s: String)
     r.hmset("hash1", Map("field1" -> Upper("val1"), "field2" -> Upper("val2")))
-    implicit val format = Format{case Upper(s) => s.toUpperCase}
-    r.hmset("hash2", Map("field1" -> Upper("val1"), "field2" -> Upper("val2")))
-    r.hmget("hash1", "field1", "field2") should be(Some(Map("field1" -> "Upper(val1)", "field2" -> "Upper(val2)")))
-    r.hmget("hash2", "field1", "field2") should be(Some(Map("field1" -> "VAL1", "field2" -> "VAL2")))
+
+    {
```
Comment (hughsimpson) [inline src/test/scala/com/redis/SerializationSpec.scala:58]: ditto

File: src/test/scala/com/redis/api/BaseApiSpec.scala [hunk 3]
```diff
@@ -261,9 +262,11 @@ trait BaseApiSpec extends FunSpec
       r.lrange("skey", 0, 10).get should equal(List(Some("1"), Some("3"), Some("10"), Some("30")))
 
       // Long serialization : return Long
-      implicit val parseLong = Parse[Long](new String(_).toLong)
-      r.sortNStore[Long]("alltest", storeAt = "skey").getOrElse(-1) should equal(4)
-      r.lrange("skey", 0, 10).get should equal(List(Some(1), Some(3), Some(10), Some(30)))
+      {
```
Comment (hughsimpson) [inline src/test/scala/com/redis/api/BaseApiSpec.scala:265]: ditto

**Comment by hughsimpson** [comment]
Not sure if it'll ever be merged 😓 that repo seems extremely quiet - doesn't seem like any contributions have been merged since 23 Jun 2019. Might be necessary to publish a fork...

**Review by debasishg — APPROVED** [review]
LGTM

**Comment by debasishg** [comment]
@hughsimpson Can u please take a look at the failed github action ? https://github.com/debasishg/scala-redis/runs/4505252759?check_suite_focus=true. The `master` is broken now ..

**Comment by hughsimpson** [comment]
Yep it's gone wonky. Not sure what happened there, I'll double-check my end but I'm fairly sure it must've been a shonky release with the docker-testkit-impl-docker-java dependency

**Pull request merged**